### ISSUE TITLE
Typings (typescript declarations) would be helpful

### ISSUE THREAD

**Issue by BurtHarris** [gulpjs/async-done#41]
The argument types and expected returns could be expressed in a typescript declarations `.d.ts` file, which would enable editors like VSCode to provide some help with using this function.   

The big restriction, re not supporting synchronous functions, might be hard to define in typescript.   Though I write in TypeScript, the API to this is complex enough that writing the definitions file for this project may be over my head.   But the expressive goodness of this should bubble up to packages like Gulp where mistakes over sync/async are more common.

**Comment by phated** [comment]
I'm actually working on a project with TypeScript right now, so I'm learning more about `.d.ts` files.  I might eventually be able to get this but I feel confident I can review a PR now.

**Comment by BurtHarris** [comment]
Let's keep this open.   I can't work on it right, but might take a stab at it.

**Comment by phated** [comment]
@demurgos want to help out with this?

**Comment by demurgos** [comment]
Yup,
I had this in mind for some time, so it's the occasion. I updated my projects last week and `async-done` is the last library I use that still does not have published types 😛. I just need to clean [the custom types I use](https://github.com/demurgos/turbo-gulp/blob/e7cb942145443746fd843fa12997618b6545e2be/src/lib/custom-typings/async-done.d.ts), I just wrote them to fix my issue and move on a few years ago so they are not ready yet. (I want to double-check with the doc and source code, add documentation and fix the signature of the `done` callback).
It shouldn't take too long: I'll send a PR on Thursday probably.

**Comment by phated** [comment]
@demurgos awesome!  I didn't realize you made turbo-gulp - have you thought of converting it to a custom registry instead of wrapping gulp? https://github.com/gulpjs/undertaker#custom-registries

**Comment by demurgos** [comment]
Yes, this is something I had in mind for some time.

I created this library about 2-3 years ago and it has gone through a lot of change. I did a big clean-up during the summer and I feel that it now reached a reasonable state. I still need to document it more because for the moment it still mostly a personal tool. My goal is to have a declarative interface to generate my gulp tasks: basically I used Gulp to create my own Grunt ;D. But it is clearly intended to be less customizable to offer a normalized complete out-of-the box experience.
I agree that it should return a custom registry: I am currently returning [an object with all the tasks](https://demurgos.github.io/turbo-gulp/interfaces/targets_lib.libtasks.html) so the design is already pretty close. I wanted to switch to custom registries but did not have time to do it yet.
Thanks for the advice.
